CAIPs/caip-295.md

```diff
@@ -0,0 +1,388 @@
+---
+caip: 295
+title: Browser Wallet Messaging for Iframes
+author: Pedro Gomes (@pedrouid)
+discussions-to: https://github.com/ChainAgnostic/CAIPs/issues/295
+status: Draft
+type: Standard
+created: 2024-06-26
+requires: 25, 27, 282
+---
+
+## Simple Summary
+
+CAIP-295 defines a standardized messaging transport for browser iframe wallets.
```

Comment:
Is there a demand for supporting iframes from the dApp side?

I'm not sure which wallets currently support iframes. I know there are some limitations on `react-native-webview` (typically used in mobile wallet dApp browser) that doesn't support injecting JS into the iframes on iOS.

Reply:
I know this is common for session based wallets and believe gnosis safe uses this approach too. I think this is going to become more common as well where iframes are going to contain the DApp logic on web2 pages kind of like what we're seeing blinks do, so it makes sense to make this stuff work with iframes when possible.